src/rules/default-import-match-filename.js

```diff
@@ -0,0 +1,206 @@
+import docsUrl from '../docsUrl'
+import isStaticRequire from '../core/staticRequire'
+import Path from 'path'
+import fs from 'fs'
+
+/**
+ * @param {string} filename
+ * @returns {string}
+ */
+function removeExtension(filename) {
+  return Path.basename(filename, Path.extname(filename))
+}
+
+/**
+ * @param {string} filename
+ * @returns {string}
+ */
+function normalizeFilename(filename) {
+  return filename.replace(/[-_.]/g, '').toLowerCase()
+}
+
+/**
+ * Test if local name matches filename.
+ * @param {string} localName
+ * @param {string} filename
+ * @returns {boolean}
+ */
+function isCompatible(localName, filename) {
+  const normalizedLocalName = localName.replace(/_/g, '').toLowerCase()
+
+  return (
+    normalizedLocalName === normalizeFilename(filename) ||
+    normalizedLocalName === normalizeFilename(removeExtension(filename))
+  )
+}
+
+/**
+ * Match 'foo' but not 'foo/bar.js' and './foo'
+ * @param {string} path
+ * @returns {boolean}
+ */
+function isBarePackageImport(path) {
+  return path !== '.' && path !== '..' && !path.includes('/')
+}
+
+/**
+ * Match paths consisting of only '.' and '..', like '.', './', '..', '../..'.
+ * @param {string} path
+ * @returns {boolean}
+ */
+function isAncestorRelativePath(path) {
+  return (
+    path.length > 0 &&
+    path[0] !== '/' &&
+    path
+      .split('/')
+      .every(segment => segment === '..' || segment === '.' || segment === '')
+  )
+}
+
+/**
+ * @param {string} packageJsonPath
+ * @returns {string | undefined}
+ */
+function getPackageJsonName(packageJsonPath) {
+  try {
+    const packageJsonContent = fs.readFileSync(packageJsonPath).toString()
```

Comment:
```suggestion
    const packageJsonContent = String(fs.readFileSync(packageJsonPath))
```
always String(x), never x.toString()